core/kernel/ldelf_syscalls.c

```diff
@@ -443,7 +468,104 @@ TEE_Result ldelf_syscall_copy_from_bin(void *dst, size_t offs, size_t num_bytes,
 	if (!binh)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	return binh_copy_to(binh, (vaddr_t)dst, offs, num_bytes);
+	buf = malloc(num_bytes);
+	if (!buf)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	res = binh_copy_to(binh, (vaddr_t)buf, offs, num_bytes);
+	if (res)
+		goto out;
+
+	res = copy_to_user(dst, buf, num_bytes);
+out:
```

Comment:
prefer instruction blocks instead of goto instruction when possible:
```c
	res = binh_copy_to(binh, (vaddr_t)buf, offs, num_bytes);
	if (!res)
		res = copy_to_user(dst, buf, num_bytes);

	free(buf);
	return res;